pygfx/helpers/_lights.py

```diff
@@ -227,4 +227,7 @@ def _update(self, transform: AffineBase):
 
         cone_length = light.distance or 1000
         cone_width = cone_length * math.tan(light.angle)
-        self.scale = (cone_width, cone_width, cone_length)
+
+        self.world.remove_callback(self.update_id)
+        self.local.scale = (cone_width, cone_width, cone_length)
+        self.update_id = self.world.on_update(self._update)
```

Comment:
Why is the callback removed and added back here?

Reply:
Changing the value of `local.scale` triggers the `world.on_update` callback, which leads to infinite recursive calls.

Reply:
Ah I see. I added a comment for future reference.